### PR TITLE
VMware: Fix dvs idempotency issue

### LINKED ISSUE #35354: Incompatible device backing specified for device
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

`vmware_guest`

##### ANSIBLE VERSION

`ansible 2.5.0 (devel e75055923b) last updated 2018/01/25 15:13:14 (GMT +200)`

##### SUMMARY

With the creation of a new VM, we are experiencing the following error message:

`Failed to create a virtual machine : Incompatible device backing specified for device 'x'.`

In the above message, x can be replaced by 1, 2, 3 etc.

We identified that this is due to change 7fbe19c. If we rebase to remove this commit, the creation of the VM succeeds as previously. If we do not assign any networks during the creation of the VM, the creation of the VM succeeds also.

##### STEPS TO REPRODUCE

Create a VM connected to a network, specifying only the network name. For example:

```yaml
  vmware_guest:
    networks:
      - name: my-network
```

##### EXPECTED RESULTS

We expect the VM to be created.

##### ACTUAL RESULTS

The VM is not created and we receive the following error message:

`Failed to create a virtual machine : Incompatible device backing specified for device 'x'.`

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fixes: #35354

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vmware_guest.py

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@cvanniekerk @pdellaert  @tchernomax  Could you please take a look ?

**Review by ghost — APPROVED** [review]

**Comment by Akasurde** [comment]
@cvanniekerk Thanks. I will wait for one more approval before merge.

**Review by tchernomax — APPROVED** [review]
LGTM

**Pull request merged**
